vignettes/datatable-reference-semantics.Rmd

```diff
@@ -258,14 +258,30 @@ head(flights)
 * The `LHS := RHS` form allows us to operate on multiple columns. In the RHS, to compute the `max` on columns specified in `.SDcols`, we make use of the base function `lapply()` along with `.SD` in the same way as we have seen before in the *"Introduction to data.table"* vignette. It returns a list of two elements, containing the maximum value corresponding to `dep_delay` and `arr_delay` for each group.
 
 #
-Before moving on to the next section, let's clean up the newly created columns `speed`, `max_speed`, `max_dep_delay` and `max_arr_delay`.
+Let's clean up the newly created columns `speed`, `max_speed`, `max_dep_delay` and `max_arr_delay`.
 
 ```{r}
 # RHS gets automatically recycled to length of LHS
 flights[, c("speed", "max_speed", "max_dep_delay", "max_arr_delay") := NULL]
 head(flights)
 ```
 
+#### -- How can we update multiple existing columns in place using `.SD`?
+
+```{r}
+char_cols <- sapply(flights, is.character)
+flights[, names(.SD) := lapply(.SD, as.factor), .SDcols = char_cols]
```

Comment:
Could also be a good place to show off `.SDcols = function` like:

```
flights[ , names(.SD) := lapply(.SD, as.factor), .SDcols = is.character]
```

Reply:
I left it in but then in the next example, I changed it to ```.SDcols = is.factor```. Sometimes a person may want to keep track of their original column classes so it could be good to show people that they have the option. 

Still, I have no heartburn if you want the first part changed to ```.SDcols = is.character``` - it would be less information overload to just do it one way throughout the example.

Reply:
Got it & makes sense, but let's reverse the order -- show the simpler way first, then the more robust way second.